### PR TITLE
Update nn-hal and integrate remote inference support

### PR DESCRIPTION
**Pull request by akodanka**

### CODE REVIEW

File: DetectionClient.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    bool get_status();
+
+private:
+    std::unique_ptr<Detection::Stub> stub_;
+    RequestDataTensors request;
+    ReplyDataTensors reply;
+    Status status;
+};
+
+#endif
```
Comment (rairatne) [inline DetectionClient.h:52]: missing new line

File: DetectionClient.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+}
+
+bool DetectionClient::get_status() {
+    if (status.ok() && (reply.data_tensors_size() > 0))
+        return 1;
+    else {
+        return 0;
+    }
+}
```
Comment (rairatne) [inline DetectionClient.cpp:127]: missing new line

File: proto/nnhal_object_detection.proto [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+// Reply message containing the Output Data Tensors(blobs)
+message ReplyDataTensors {
+  repeated DataTensor data_tensors = 1;
+}
+
+// Request message containing the Input Data Tensors(blobs)
+message RequestDataTensors {
+  repeated DataTensor data_tensors = 1;
+}
```
Comment (rairatne) [inline proto/nnhal_object_detection.proto:60]: missing new line

**Review by rairatne — APPROVED** [review]
LGTM apart from mentioned comments!!

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**